### PR TITLE
Move database and similar packages to components

### PR DESCRIPTION
**Pull request by rynowak**
# Description

This change moves and rearranges some our packages so they are easy to reference. The new `components` folder is for shared components that are used in all of our code. Previously these components were part of UCP which doesn't make sense, they are used by all of our control-plane services.

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

## Contributor checklist
Please verify that the PR meets the following requirements, where applicable:

- [ ] An overview of proposed schema changes is included in a linked GitHub issue.
- [ ] A design document PR is created in the [design-notes repository](https://github.com/radius-project/design-notes/), if new APIs are being introduced.
- [ ] If applicable, design document has been reviewed and approved by Radius maintainers/approvers.
- [ ] A PR for the [samples repository](https://github.com/radius-project/samples) is created, if existing samples are affected by the changes in this PR.
- [ ] A PR for the [documentation repository](https://github.com/radius-project/docs) is created, if the changes in this PR affect the documentation or any user facing updates are made.
- [ ] A PR for the [recipes repository](https://github.com/radius-project/recipes) is created, if existing recipes are affected by the changes in this PR.

### CODE REVIEW

File: bicep-types
Comment (ytimocin) [inline bicep-types:1]: do we need this?

File: pkg/recipes/terraform/execute.go [hunk 1]
```diff
@@ -27,13 +27,13 @@ import (
 	install "github.com/hashicorp/hc-install"
 	"github.com/hashicorp/terraform-exec/tfexec"
 	tfjson "github.com/hashicorp/terraform-json"
+	ucp_provider "github.com/radius-project/radius/pkg/components/secret/secretprovider"
```
Comment (ytimocin) [inline pkg/recipes/terraform/execute.go:30]: should we remove the alias?

File: bicep-types
Comment (rynowak) [reply bicep-types:1]: Oops, this should not be changing in this PR.

File: pkg/recipes/terraform/execute.go (hunk 1 above)
Comment (rynowak) [reply pkg/recipes/terraform/execute.go:30]: Sure, I'll clean this up too.

**Review by ytimocin — DISMISSED** [review]

**Comment by rynowak** [comment]
Working on fixing `make generate` error

**Review by ytimocin — APPROVED** [review]

**Review by lakshmimsft — APPROVED** [review]

**Comment by rynowak** [comment]
Test failure was a 409 conflict upstream in Azure. Unrelated to changes in this PR. I'll try re-running and then investigate if it repros.

**Comment by rynowak** [comment]
Going to bypass the functional test behavior because this change is a refactor, and this appears to be a transient Azure issue. Let's keep an eye on this, because it will likely fail in other PRs if its not resolved soon.

**Pull request merged**